java-checks/src/main/java/org/sonar/java/checks/SwitchCaseWithoutBreakCheck.java

```diff
@@ -54,14 +61,50 @@ public void visitSwitchStatement(SwitchStatementTree switchStatement) {
         CaseLabelTree caseLabel = caseGroup.labels().get(caseGroup.labels().size() - 1);
 
           // Reverse the body as commonly the unconditional exit will be at the end of the body.
-        if (Lists.reverse(caseGroup.body()).stream().noneMatch(SwitchCaseWithoutBreakCheck::isUnconditionalExit)) {
+        if (Lists.reverse(caseGroup.body()).stream().noneMatch(SwitchCaseWithoutBreakCheck::isUnconditionalExit) && !intentionalFallThrough(switchStatement, caseGroup)) {
           context.reportIssue(this, caseLabel, "End this switch case with an unconditional break, return or throw statement.");
         }
       });
 
     super.visitSwitchStatement(switchStatement);
   }
 
+  private boolean intentionalFallThrough(SwitchStatementTree switchStatement, CaseGroupTree caseGroup) {
+    FallThroughCommentVisitor visitor = new FallThroughCommentVisitor();
+    // Check first token of next case group when comment is last element of case group it is attached to next group.
+    CaseGroupTree nextCaseGroup = switchStatement.cases().get(switchStatement.cases().indexOf(caseGroup) + 1);
+    List<Tree> treesToScan = ImmutableList.<Tree>builder().addAll(caseGroup.body()).add(nextCaseGroup.firstToken()).build();
+    visitor.scan(treesToScan);
+    return visitor.hasComment;
+  }
+
+  private static class FallThroughCommentVisitor extends SubscriptionVisitor {
+
+    private static final Pattern FALL_THROUGH_PATTERN = Pattern.compile("fall\\-?through", Pattern.CASE_INSENSITIVE);
```

Comment:
I have seen the comment also written with a `s`  (`"falls-through"`), or with another spelling (`"fallthru"`). I think it should be handle as well.
For instance: https://eslint.org/docs/rules/no-fallthrough

Examples of what is supported elsewhere:
- https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/
- http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.html